server/queue/queue-jms/src/main/java/org/apache/james/queue/jms/ConsumerOptions.java

```diff
@@ -0,0 +1,42 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.queue.jms;
+
+import org.apache.james.queue.api.MailQueue;
+
+/**
+ * Provides additional options when creating consumers.
+ */
+public interface ConsumerOptions {
+    /**
+     * The empty consumer options.
+     */
+    static ConsumerOptions empty() {
```

Comment:
as it's immutable, you can very well write : 
```
ConsumerOptions EMPTY = Function.identity();
```

Reply:
Do you suggest to use field instead of method?

Reply:
I would prefer to use method (especially having a successful examples of Optional.empty(), Function.identity(), etc).

What reasoning you have on that?

Reply:
The rational is : it never changes so you can easily avoid useless object creation.
The reason why Optional and Function uses methods for that is generic handling.
see for example Optional : 
```
public static<T> Optional<T> empty() {
        @SuppressWarnings("unchecked")
        Optional<T> t = (Optional<T>) EMPTY;
        return t;
    }
```
It allows to return the right Optional<T> type for the caller. They however use a singleton anyway for the same reason I asked it.
In our case, the type is not generic so we don't need this extra complexity but if you want to return a singleton with a method, I don't have anything against it.

Reply:
Sounds very logical. Surely I would like to use method approach and return singleton instance through, but in our case we must choose between field and method. I think for this particular case field is better approach.

Thanks for details.